crates/net/network/src/transactions/validation.rs

```diff
@@ -22,7 +23,13 @@ pub trait ValidateTx68 {
     /// entry. Returns [`ValidationOutcome`] which signals to the caller wether to fetch the
     /// transaction or wether to drop it, and wether the sender of the announcement should be
     /// penalized.
-    fn should_fetch(&self, ty: u8, hash: &TxHash, size: usize) -> ValidationOutcome;
+    fn should_fetch(
+        &self,
+        ty: u8,
+        hash: &TxHash,
+        size: usize,
+        tx_types_counter: &mut TxTypesCounter,
```

Comment:
hmm, should the caller be responsible for updating this?

the update can be derived from the return value?